src/tag_index.c

```diff
@@ -55,13 +55,18 @@ char *TagIndex_SepString(char sep, char **s, size_t *toklen) {
   return start;
 }
 
-/* Preprocess a document tag field, returning a vector of all tags split from the content */
-char **TagIndex_Preprocess(char sep, TagFieldFlags flags, const DocumentField *data) {
+int tokenizeTagString(RedisModuleString *str, char sep, TagFieldFlags flags, char ***resArray) {
   size_t sz;
-  char *p = (char *)RedisModule_StringPtrLen(data->text, &sz);
-  if (!p || sz == 0) return NULL;
-  char **ret = array_new(char *, 4);
-  char *pp = p = rm_strndup(p, sz);
+  char *p = (char *)RedisModule_StringPtrLen(str, &sz);
+  if (!p || sz == 0) return REDISMODULE_ERR;
+
+  // remove quotes on JSON
+  if (flags & TagField_RemoveQuotes && *p == '"' && *(p + sz -1) == '"') {
```

Comment:
If you would have a check that the value is a string you would not have need that ...

Reply:
We do not know if the source is a hash or json.

Reply:
If you take a String from the json it would have come without the `"`